lib/Echidna/Transaction.hs

```diff
@@ -224,3 +229,36 @@ setupTx (Tx c s r g gp v (t, b)) = liftSH . sequence_ $
       SolCalldata cd  -> loadContract r >> state . calldata .= cd
     encode (n, vs) = abiCalldata
       (encodeSig (n, abiValueType <$> vs)) $ V.fromList vs
+
+saveTxs :: Maybe FilePath -> [[Tx]] -> IO ()
+saveTxs (Just d) txs = mapM_ (\v -> do let fn = d ++ "/" ++ (show . hash . show) v ++ ".txt"
+                                       b <- doesFileExist fn
+                                       unless b $ encodeFile fn (sv v)
+                             ) txs
+                             where sv = toJSON
+saveTxs Nothing  _   = return ()
+
+listDirectory :: FilePath -> IO [FilePath]
+listDirectory path = Prelude.filter f <$> getDirectoryContents path
+  where f filename = filename /= "." && filename /= ".."
+
+withCurrentDirectory :: FilePath  -- ^ Directory to execute in
+                     -> IO a      -- ^ Action to be executed
+                     -> IO a
+withCurrentDirectory dir action =
+  bracket getCurrentDirectory setCurrentDirectory $ \_ -> do
+    setCurrentDirectory dir
+    action
+
+loadTxs :: Maybe FilePath -> IO [[Tx]]
+loadTxs (Just d) = do 
+  fs <- listDirectory d
+  xs <- mapM makeRelativeToCurrentDirectory fs
+  mtxs <- withCurrentDirectory d (mapM readCall xs)
+  let txs = catMaybes mtxs
+  putStrLn ("Loaded total of " ++ show (length xs) ++ " transactions from " ++ d)
+  return txs
+  where readCall f = do !buf <- LBS.readFile f
+                        return (decode buf :: Maybe [Tx]) 
```

Comment:
we can use `decodeStrict` with `LBS.fromStrict` (also change this to a standard non-qualified import). also i think the type checker should deduce `Maybe [Tx]` from `catMaybes`.

Reply:
if i were to rewrite this function it would look something like `readCall f = decode . fromStrict <$> BS.readFile f`.

Reply:
Fixed.